plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/SecurityUtils.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.repositories.hdfs;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.plugin.hadoop.hdfs.HdfsPlugin;
+
+import java.io.IOException;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+
+public class SecurityUtils {
+
+    public static <V> V execute(FileSystemFactory ffs, FsCallback<V> callback) throws IOException {
+        return execute(ffs.getFileSystem(), callback);
+    }
+
+    public static <V> V execute(FileSystem fs, FsCallback<V> callback) throws IOException {
+        try {
```

Comment:
this method (and a few others here) are security holes without a check. basically, before entering any `doPrivileged` block, please do a check for `org.elasticsearch.SpecialPermission`, see https://www.elastic.co/guide/en/elasticsearch/plugins/2.x/plugin-authors.html#_java_security_permissions

if we just add the check like that consistently and religiously, we just don't have to think about it: we won't have privilege escalation problems.

otherwise without it, if a script for example, gets ahold of this guy and invokes it, we will "forget its a script" because we truncate the stack: then the script can potentially do things it should not be able to do. I hope that makes sense.